### PR TITLE
rebase: update kubernetes deps to 1.27.1

### LINKED ISSUE #3743: Update Kubernetes dependencies/CI to 1.27
Kubernetes 1.27 is released https://kubernetes.io/blog/2023/04/11/kubernetes-v1-27-release/ let's make use of it in cephcsi

- [x]  Make Kubernetes E2E test mandatory in CI Run
- [x] Update Mergify rules to consider tests on kube 1.27
- [x] go dependency update
- [x] Readme update

### PR DESCRIPTION
**Pull request by riya-singhal31**
As the k8s 1.27 is released, updating the do dependencies to 1.27.

### CODE REVIEW

**Comment by Madhu-1** [comment]
@riya-singhal31 please rebase the PR

**Comment by riya-singhal31** [comment]
> @riya-singhal31 please rebase the PR

Rebased, thanks!

File: go.mod [hunk 1]
```diff
@@ -35,21 +35,24 @@ require (
 	//
 	// when updating k8s.io/kubernetes, make sure to update the replace section too
 	//
-	k8s.io/api v0.26.3
-	k8s.io/apimachinery v0.26.3
+	k8s.io/api v0.27.0
```
Comment (Madhu-1) [inline go.mod]: we have kubernetes 1.27.1 release , instead of updating to older release lets update to new release.

**Comment by Madhu-1** [comment]
https://github.com/ceph/ceph-csi/blob/f0e6ee047ee388a7299d70dfbb5784546540725f/go.mod#L183-L186 time to remove this as we are moving to 1.27?

File: go.mod (hunk 1 above)
Comment (riya-singhal31) [reply go.mod]: Yes, sure.
Updated to 1.27.1

**Comment by riya-singhal31** [comment]
> https://github.com/ceph/ceph-csi/blob/f0e6ee047ee388a7299d70dfbb5784546540725f/go.mod#L183-L186
> 
> time to remove this as we are moving to 1.27?

Removed the other two, leaving github.com/portworx/sched-ops => github.com/portworx/sched-ops v0.20.4-openstorage-rc3 
as this is still required.

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
Please fix the CI failures

**Comment by Madhu-1** [comment]
@riya-singhal31 please rebase the PR

**Comment by riya-singhal31** [comment]
> @riya-singhal31 please rebase the PR

Rebased, thanks!

**Comment by riya-singhal31** [comment]
It seems like a bug in operator-sdk client.go https://github.com/operator-framework/operator-sdk/issues/6396

**Comment by riya-singhal31** [comment]
Thanks for the commits @nixpanic. I'll take a look and fix the rest of CIs.

**Comment by Madhu-1** [comment]
@riya-singhal31 can you please refresh the PR?

**Comment by riya-singhal31** [comment]
> @riya-singhal31 can you please refresh the PR?

Thanks, rebased.
Fixing the linters.

**Pull request closed without merging**